adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceBannerAd.m

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterIronSourceBannerAdDelegate.h"
+#import "GADMAdapterIronSourceBannerAd.h"
+#import "GADMAdapterIronSourceConstants.h"
+#import "GADMAdapterIronSourceUtils.h"
+#import "GADMediationAdapterIronSource.h"
+
+@interface GADMAdapterIronSourceBannerAd ()
+
+// The completion handler to call when the ad loading succeeds or fails.
+@property(copy, nonatomic) GADMediationBannerLoadCompletionHandler bannerAdLoadCompletionHandler;
+
+// An ad event delegate to invoke when ad rendering events occur.
+@property(weak, nonatomic) id<GADMediationBannerAdEventDelegate> bannerAdEventDelegate;
+
+/// Holds the ID of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceID;
+
+/// Holds the state of the ad instance to be presented.
+@property(nonatomic, copy) NSString *instanceState;
```

Comment:
Can you clarify why we need to keep the state of the ad instance in the ironSource SDK perspective?

Reply:
Thanks @joshh-devrel 
IronSource SDK doesn't support loading an instance while it is in the process of loading.
Therefore we keep the state of the instance to avoid duplicate loading